dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Register/EmailConfirmation.cshtml.cs

```diff
@@ -46,4 +47,14 @@ public async Task<IActionResult> OnPost()
 
         return Redirect(_linkGenerator.RegisterPhone());
     }
+
+    public override void OnPageHandlerExecuting(PageHandlerExecutingContext context)
+    {
+        var authenticationState = context.HttpContext.GetAuthenticationState();
+
+        if (!authenticationState.EmailAddressSet)
```

Comment:
Added this to email confimation as it seemed sensible even though we are not quite sure how to measure milestones in this journey